### PR TITLE
Feature/97 register info ios

### LINKED ISSUE #97: Register page info slides iOS

### PR DESCRIPTION
**Pull request by Yarin18**

### CODE REVIEW

File: iosApp/iosApp/components/PagerView.swift
Comment (DanneD-Wisemen) [inline iosApp/iosApp/components/PagerView.swift:1]: THis looks like its missing some indentations :)

File: iosApp/iosApp/components/PagerView.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ init(_ data: Data,
+        currentIndex: Binding<Int>,
+        @ViewBuilder content: @escaping (Data.Element) -> Content) {
+    self.data = data
+    _currentIndex = currentIndex
+    self.content = content
+ }
+
+ var body: some View {
+    GeometryReader { geometry in
```
Comment (DanneD-Wisemen) [inline iosApp/iosApp/components/PagerView.swift]: Just for some information, what does this GeometryReader do?

File: iosApp/iosApp/components/RegisterScreenComponents.swift [hunk 2]
```diff
@@ -90,3 +89,153 @@ struct PrivacySwitch: View {
...
+    
+    var body: some View {
+        VStack(spacing: 0) {
+            Spacer()
+            InfoTitle(title: strings.get(id: SharedRes.strings().level_system, args: []))
+                .padding(.bottom, 18)
+            Image("progress_bar")
+                .padding(.bottom, 56)
+                .frame(width: 294, height: 41)
+            Text(strings.get(id: SharedRes.strings().level_system_title, args: []))
```
Comment (DanneD-Wisemen) [inline iosApp/iosApp/components/RegisterScreenComponents.swift]: Add a white line between components

File: iosApp/iosApp/components/RegisterScreenComponents.swift
Comment (DanneD-Wisemen) [inline iosApp/iosApp/components/RegisterScreenComponents.swift:1]: Applicable for the whole file, add some whitespacing between parts

File: iosApp/iosApp/ui/pages/HomeScreen.swift [hunk 3]
```diff
@@ -5,6 +5,7 @@ struct HomeScreen: View {
     var user = FirebaseRepository().getCurrentUserUid()
     
     var body: some View {
+        Text("HomeScreen")
```
Comment (DanneD-Wisemen) [inline iosApp/iosApp/ui/pages/HomeScreen.swift]: Hardcoded text

File: iosApp/iosApp/ui/pages/RegistrationScreen.swift [hunk 4]
```diff
@@ -29,38 +29,36 @@ struct RegistrationScreen: View {
                 PermissionStep { action in
                     viewModel.onAction(action: action)
                 }
+            } else if (currentRegisterState == CurrentRegisterState.info) {
```
Comment (DanneD-Wisemen) [inline iosApp/iosApp/ui/pages/RegistrationScreen.swift]: See previous comments on this block

File: iosApp/iosApp/ui/pages/RegistrationScreen.swift [hunk 5]
```diff
@@ -181,3 +179,67 @@ struct PermissionStep: View {
...
+    var body: some View {
+        ZStack {
+            PagerView(pages, currentIndex: $currentIndex) { page in
+                switch page {
+                case 1: InfoPageCharging()
+                case 2: InfoPageChargingRules()
+                case 3: InfoPageLevelInfo()
+                case 4: InfoPageLevelingInfo()
+                default:
+                    Text("This should not happen")
```
Comment (DanneD-Wisemen) [inline iosApp/iosApp/ui/pages/RegistrationScreen.swift]: The app should crash in this place, try throwing an fatalerror()

**Review by DanneD-Wisemen — CHANGES_REQUESTED** [review]

File: iosApp/iosApp/ui/pages/RegistrationScreen.swift (hunk 5 above)
Comment (DanneD-Wisemen) [reply iosApp/iosApp/ui/pages/RegistrationScreen.swift]: https://cocoacasts.com/what-is-fatalerror-in-swift-and-when-to-use-it

File: iosApp/iosApp/components/PagerView.swift (hunk 1 above)
Comment (Yarin18) [reply iosApp/iosApp/components/PagerView.swift]: as far as I understand it its a tool that allows you to get information about the position and size of certain components which can be used to dynamic,ically calculate size of things

File: iosApp/iosApp/ui/pages/RegistrationScreen.swift (hunk 5 above)
Comment (DanneD-Wisemen) [reply iosApp/iosApp/ui/pages/RegistrationScreen.swift]: 👆🏻

File: iosApp/iosApp/ui/pages/RegistrationScreen.swift [hunk 6]
```diff
@@ -21,43 +22,42 @@ struct RegistrationScreen: View {
...
         VStack() {
-            EmailTextField(
-                email:
-                    Binding(
-                        get: { viewModel.getState().email },
-                        set: { value in onAction(RegisterScreenUiAction.OnEmailChangedAction(email: value)) }
-                    ),
-                isValid: viewModel.getState().isEmailValid,
-                clearInputAction: { onAction(RegisterScreenUiAction.OnEmailChangedAction(email: "")) }
+            EmailTextField(email:
```
Comment (DanneD-Wisemen) [inline iosApp/iosApp/ui/pages/RegistrationScreen.swift]: weird indentation

File: iosApp/iosApp/ui/pages/RegistrationScreen.swift [hunk 7]
```diff
@@ -21,43 +22,42 @@ struct RegistrationScreen: View {
...
+                           isValid: state.isEmailValid,
+                           clearInputAction: { onAction(RegisterScreenUiAction.OnEmailChangedAction(email: "")) }
+                           
             )
-            Text(strings.get(id: SharedRes.strings().privacy, args: []))
-                .alignmentGuide(.leading) { _ in 0 }
-                .frame(maxWidth: .infinity, alignment: .leading)
-                .padding(.leading, 16)
-            NextButton(state: viewModel.getState(), onAction: onAction)
+            .alignmentGuide(.leading) { _ in 0 }
```
Comment (DanneD-Wisemen) [inline iosApp/iosApp/ui/pages/RegistrationScreen.swift:57]: check indentation

File: iosApp/iosApp/ui/pages/RegistrationScreen.swift [hunk 8]
```diff
@@ -179,3 +178,69 @@ struct PermissionStep: View {
...
+    var body: some View {
+        ZStack {
+            PagerView(pages, currentIndex: $currentIndex) { page in
+                switch page {
+                case 1: InfoPageCharging()
+                case 2: InfoPageChargingRules()
+                case 3: InfoPageLevelInfo()
+                case 4: InfoPageLevelingInfo()
+                default:
+                    Text("This should not happen")
```
Comment (DanneD-Wisemen) [inline iosApp/iosApp/ui/pages/RegistrationScreen.swift]: fatalError() would be better

**Review by DanneD-Wisemen — CHANGES_REQUESTED** [review]

**Review by DanneD-Wisemen — APPROVED** [review]

**Pull request merged**